src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -1046,11 +1047,28 @@ public static SyntaxKind GetPreprocessorKeywordKind(string text)
             }
         }
 
+        private static SyntaxKind[] _lazyContextualKeywordKindsArray;
```

Comment:
@CyrusNajmabadi We want the result to be read-only, even if the user casts it to a mutable collection type. This is part of the Roslyn public API.